/* 
	Copyright (c) 2018 Azeez Adewale <azeezadewale98@gmail.com"> 
	MIT License Copyright (c) 2018 simple 

*/

/* 
 * @filename - Throw.sim
 * @author - Azeez Adewale
 * @date - 18 Febuary 2018
 */
 
module simple.debugging

call "../core/loadcoredymodule.sim"
 
/*
	Throw an error at any point in the program with this block
	the parameter @Bold errorvalue @End is the main error message
	display when the error is thrown. Calling this block ends the 
	program after the error has been displayed. The error is 
	thrown in the format bellow
	@Code
	
		Line 24 -> Unknown Error Occur
			at throw() in file Throw.sim
			at line 40 at three() in file debugging.sim
			at line 37 at two() in file debugging.sim
			at line 34 at one() in file debugging.sim
			at line 29 at main() in file debugging.sim
			at line 21 in file debugging.sim
			
	@EndCode
*/
block throw(errorvalue)
	if errorvalue = "" {
		errorvalue = "Unknown Error Occur"
	}
	__throw(errorvalue)
	__exit__
	
	